### PR TITLE
CI: Port MSYS2 jobs from appveyor to github actions

### PR DESCRIPTION
**Pull request by lazka**
This is motivated by our new github action https://github.com/msys2/setup-msys2 which makes spawning a MSYS2 env easy and hides the install/update details..

(and also github actions has 20 parallel jobs.. compared to 1 in appveyor)

### CODE REVIEW

File: .github/workflows/msys2-ci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-ninja
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-pkg-config
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-python3
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-python3-pip
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-ragel
+      - name: Install extra dependencies
+        run: |
+          pip install fonttools
+          pip install --upgrade meson==0.53.0
+          curl "https://raw.githubusercontent.com/mirror/mingw-w64/023eb04c396d4e8d8fcf604cfababc53dae13398/mingw-w64-headers/include/dwrite_1.h" > "$MINGW_PREFIX/$MINGW_CHOST/include/dwrite_1.h"
```
Comment (ebraminio) [inline .github/workflows/msys2-ci.yml:48]: This probably can be dropped as your are building hb-directwrite in msys2 successfully now.

**Comment by ebraminio** [comment]
Looks great! Thank you so much!

File: .github/workflows/msys2-ci.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        run: |
+          meson build \
+            --wrap-mode=nofallback \
+            -Dfreetype=enabled \
+            -Dglib=enabled \
+            -Dcairo=enabled \
+            -Dgobject=enabled \
+            -Dgdi=enabled \
+            -Ddirectwrite=enabled \
+            -Dgraphite=enabled
```
Comment (ebraminio) [inline .github/workflows/msys2-ci.yml:59]: Guess now can be reduced to `--auto-features=enabled -Dgraphite=enabled`

File: .github/workflows/msys2-ci.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-ragel
+      - name: Install extra dependencies
+        run: |
+          pip install fonttools
+          pip install --upgrade meson==0.53.0
+          curl "https://raw.githubusercontent.com/mirror/mingw-w64/023eb04c396d4e8d8fcf604cfababc53dae13398/mingw-w64-headers/include/dwrite_1.h" > "$MINGW_PREFIX/$MINGW_CHOST/include/dwrite_1.h"
+      - name: Build
+        run: |
+          meson build \
+            --wrap-mode=nofallback \
```
Comment (ebraminio) [inline .github/workflows/msys2-ci.yml:52]: Guess `--wrap-mode=nodownload` will be even better.

File: .github/workflows/msys2-ci.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-icu
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-ninja
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-pkg-config
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-python3
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-python3-pip
+            mingw-w64-${{ matrix.MSYS2_ARCH }}-ragel
+      - name: Install extra dependencies
+        run: |
+          pip install fonttools
+          pip install --upgrade meson==0.53.0
```
Comment (ebraminio) [inline .github/workflows/msys2-ci.yml:47]: We are okay with mingw's meson and even fonttools, I didn't do that as for some reason I weren't getting an updated version with previous, the shiny new setup shouldn't have the issues I'd guess.

**Comment by ebraminio** [comment]
Just to note your CI config is already cleaner than most of our current CI configurations, thank you so much for putting time for this :)

File: .github/workflows/msys2-ci.yml (hunk 1 above)
Comment (lazka) [reply .github/workflows/msys2-ci.yml]: ok, done

File: .github/workflows/msys2-ci.yml (hunk 4 above)
Comment (lazka) [reply .github/workflows/msys2-ci.yml]: ok, switched to the msys2 packages

File: .github/workflows/msys2-ci.yml (hunk 2 above)
Comment (lazka) [reply .github/workflows/msys2-ci.yml]: done, I had to disable benchmark though since that failed.

**Comment by lazka** [comment]
Added the suggestions in a second commit, to separate the porting from the cleanup part.

Should be ready.

**Comment by ebraminio** [comment]
Fantastic. Thank you so much :)

File: .github/workflows/msys2-ci.yml (hunk 2 above)
Comment (ebraminio) [reply .github/workflows/msys2-ci.yml]: have applied some workaround and removed it, just am notifying so you don't need to add it to msys2 pack also :)

**Pull request merged**